src/index.js

```diff
@@ -26,44 +43,53 @@ const clearField = () => {
-  const template = document.querySelector('#list-item-template');
-  const templateClone = template.content.cloneNode(true);
-  const taskContent = templateClone.querySelector('[data-list-item-text]');
-  const checkBox = templateClone.querySelector('[data-list-item-checkbox]');
-
-  // Create a new Todo
   const todoTemplate = new Todo(todoList.length, inputField.value, false);
   todoList.push(todoTemplate);
-  saveList();
-  taskContent.innerText = inputField.value;
-  if (checkBox.checked) {
-    taskContent.classList.add('.line-through');
-    checkBox.checked = true;
-  } else {
-    taskContent.classList.remove('.line-through');
-    checkBox.checked = false;
-  }
-  todoListContainer.appendChild(templateClone);
+  renderTodo(todoTemplate);
+  // saveList();
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
   clearField();
 });
 
-todoList.forEach((todo) => {
-  const template = document.querySelector('#list-item-template');
-  const templateClone = template.content.cloneNode(true);
-  const taskContent = templateClone.querySelector('[data-list-item-text]');
-  const checkBox = templateClone.querySelector('[data-list-item-checkbox]');
-  checkBox.addEventListener('change', () => {
-    // Save checkbox event in local storage
-    checkBoxEvent(todo, checkBox, taskContent, saveList);
-  });
-  taskContent.innerText = todo.description;
-  checkBox.checked = todo.completed;
-  saveList();
-  if (checkBox.checked) {
-    taskContent.classList.add('line-through');
-  } else {
-    taskContent.classList.remove('line-through');
+todoListContainer.addEventListener('click', (e) => {
+  if (!e.target.matches('[data-button-delete]')) return;
+
+  const parent = e.target.closest('.list-item');
+  const todoIndex = parseInt(parent.dataset.todoIndex);
+  parent.remove(); // removes from the screen
+  todoList = todoList.filter((todo) => todo.index !== todoIndex); // removes from the list
+  // saveList(); saves updated list
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+});
+
+clearAllCompleted.addEventListener('click', () => {
+  todoList = todoList.filter((todo) => todo.completed === false);
+  // saveList();
+  localStorage.setItem(TODO_LIST_KEY, JSON.stringify(todoList));
+  location.reload();
+});
+
+const index = () => {
+  let length = todoList.length;
+  for (let i = 0; i < length; i++) {
+    todo[i].index = i;
   }
-  todoListContainer.appendChild(templateClone);
+};
+
+todoListContainer.addEventListener('click', (e) => {
+  if (!e.target.matches('[data-button-edit]')) return;
+
+  const parent = e.target.closest('.list-item');
+  const todoId = parent.dataset.todoIndex;
+  const todoIndex = parent.querySelector('[data-list-item-text]');
+  console.log('index: ', todoIndex);
+  console.log('ID: ', todoId);
+  const editedTodo = prompt('Please edit your todo', '');
+  if (editedTodo != null) {
+    todoList[todoId].description = editedTodo;
+  }
+  // index();
+  // saveList();
```

Comment:
- [ ] Kindly remove all commented code from your code base to make it look cleaner.

Reply:
Done ✅